spec/models/program_session_spec.rb

```diff
@@ -1,5 +1,134 @@
-  pending "add some examples to (or delete) #{__FILE__}"
+describe ProgramSession do
+
+  let(:proposal) { create(:proposal_with_track, :with_two_speakers) }
+
+  it "responds to .create_from_proposal" do
+    expect(ProgramSession).to respond_to(:create_from_proposal)
+  end
+
+  describe "#create_from_proposal" do
+
+    it "creates a new program session with the same title as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.title).to eq(proposal.title)
+    end
+
+    it "creates a new program session with the same abstract as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.abstract).to eq(proposal.abstract)
+    end
+
+    it "creates a new program session with the same event as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.event_id).to eq(proposal.event_id)
+    end
+
+    it "creates a new program session with the same session format as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.session_format_id).to eq(proposal.session_format_id)
+    end
+
+    it "creates a new program session with the same track as the given proposal" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.track_id).to eq(proposal.track_id)
+    end
+
+    it "creates a program session that has the proposal id" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.proposal_id).to eq(proposal.id)
+    end
+
+    it "creates a program session that is active" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      expect(session.state).to eq("active")
+    end
+
+    it "sets program session id for all speakers" do
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.program_session_id).to eq(session.id)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "sets speaker_name from user on each speaker" do
+      proposal.speakers.each do |speaker|
+        expect(speaker.speaker_name).to eq(nil)
+      end
+
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.speaker_name).to eq(speaker.user.name)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "sets speaker_email from user on each speaker" do
+      proposal.speakers.each do |speaker|
+        expect(speaker.speaker_email).to eq(nil)
+      end
+
+      session = ProgramSession.create_from_proposal(proposal)
+
+      session.speakers.each do |speaker|
+        expect(speaker.speaker_email).to eq(speaker.user.email)
+        expect(speaker.changed?).to be(false)
+      end
+    end
+
+    it "does not overwrite name while simultaneously setting email" do
```

Comment:
This reads a little ambiguously to me. Makes me wonder if not overwriting the name has something to do with setting the email? Since each field is set independently of the other, and are being tested here independently of one another, it's enough to say "does not overwrite speaker_name if it already has a value".